packages/pg-gateway/test/deno/tcp.test.ts

```diff
@@ -0,0 +1,53 @@
+// @deno-types="npm:@types/pg"
+import pg from 'npm:pg';
+
+import { expect } from 'jsr:@std/expect';
+import { afterAll, beforeAll, describe, it } from 'jsr:@std/testing/bdd';
+import { PGlite } from 'npm:@electric-sql/pglite';
+import { fromDenoConn } from 'pg-gateway/deno';
+
+const { Client } = pg;
+let listener: Deno.TcpListener;
+
+async function startServer(listener: Deno.TcpListener) {
+  for await (const conn of listener) {
+    const db = new PGlite();
+
+    await fromDenoConn(conn, {
+      async onStartup() {
+        await db.waitReady;
+      },
+      async onMessage(data, { isAuthenticated }) {
+        if (!isAuthenticated) {
+          return;
+        }
+        return await db.execProtocolRaw(data);
+      },
+    });
+  }
+}
+
+async function connect() {
+  const client = new Client('postgresql://postgres:postgres@localhost:54320/postgres');
+  await client.connect();
+  return client;
+}
+
+beforeAll(() => {
+  listener = Deno.listen({ port: 54320 });
+  startServer(listener);
+});
+
+afterAll(() => {
+  listener.close();
+});
+
+describe('pglite', () => {
+  it('simple query returns result', async () => {
+    const client = await connect();
+    const res = await client.query("select 'Hello world!' as message");
+    const [{ message }] = res.rows;
+    expect(message).toBe('Hello world!');
+    await client.end();
+  });
+});
```

Comment:
Disposable resource detected, `using` to the rescue! 😄

I find it super useful in tests to make the tests more readable and don't have to think about disposing the resources anymore!

I'll let you make the changes in the other tests if you think it makes sense. 👍 

```suggestion
async function getClient(databaseUrl: string) {
  const client = new Client(databaseUrl);
  await client.connect();
  return {
    client,
    async [Symbol.asyncDispose]() {
      await client.end();
    },
  }
}

beforeAll(() => {
  listener = Deno.listen({ port: 54320 });
  startServer(listener);
});

afterAll(() => {
  listener.close();
});

describe('pglite', () => {
  it('simple query returns result', async () => {
    await using db = await getClient('postgresql://postgres:postgres@localhost:54320/postgres');
    const res = await client.query("select 'Hello world!' as message");
    const [{ message }] = res.rows;
    expect(message).toBe('Hello world!');
  });
});
```

Reply:
Great call, I love using `using` 😄 will update

Reply:
Updated [in this commit](https://github.com/supabase-community/pg-gateway/pull/11/commits/b3efdad3281b4c3649b3919a09efe76122aaeed7)